src/backend.py

```diff
@@ -0,0 +1,726 @@
+
+@add_route("/api/game/active_config")
+def app_game_config_filename(request):
+    return json_dumps({
+        "active_config":ds_read_record("configuration", 0)["gamename"]
+    }), 200
+
+@add_route("/api/game/configs_list")
+def app_game_configs_list(request):
+    from GameDefsLoad import list_game_configs
+    return json_dumps(list_game_configs()), 200
+        
+
+#
+# Memory
+#
+@add_route("/api/memory/reset", auth=True)
+def app_reset_memory(request):
+    import reset_control
+    reset_control.reset()
+    sleep(2)
+    blank_ram()
+    sleep(1)
+    reset_control.release(True)
+    server.reset_bootup_counters()
+    
+
+@add_route("/api/memory-snapshot")
+def download_memory(request):
+    # Stream memory values directly to the response to save RAM
+    def memory_values_generator():
+        for value in ram_access:
+            yield f"{value}\n".encode('utf-8')               
+
+    headers = {
+        'Content-Type': 'text/plain',
+        'Content-Disposition': 'attachment; filename=memory.txt',
+        'Connection': 'close'
+    }
+    
+    return memory_values_generator(), 200, headers
+
+   
+#
+# Leaderboard
+#
+def get_scoreboard(key):
+    '''Get the leaderboard from memory'''
+    rows = []
+    for i in range(ds_memory_map[key]["count"]):
+        row = ds_read_record(key, i)
+        if row.get("score", 0) > 0:
+            rows.append(row)
+    
+    # sort the rows by score
+    rows.sort(key=lambda x: x["score"], reverse=True)
+    
+    # add the rank to each row
+    for i, row in enumerate(rows):
+        row["rank"] = i + 1
+
+    return json_dumps(rows), 200
+
+@add_route("/api/leaders")
+def app_leaderBoardRead(request):
+    return get_scoreboard("leaders")
+
+@add_route("/api/tournament")
+def app_tournamentRead(request):
+    return get_scoreboard("tournament")
+
+@add_route("/api/leaders/reset", auth=True)
+def app_resetScores(request):
+    from SPI_DataStore import blankStruct
+    blankStruct("leaders")
+
+@add_route("/api/tournament/reset", auth=True)
+def app_tournamentClear(request):
+    from SPI_DataStore import blankStruct
+    import SharedState
+    blankStruct("tournament")
+    SharedState.gameCounter=0
+
+#
+# Players
+#
+@add_route("/api/players")
+def app_getPlayers(request):
+    players = {}
+    count = ds_memory_map["names"]["count"]
+    # Iterate through the player records
+    for i in range(count):
+        record = ds_read_record("names", i)
+        initials = record['initials'].replace('\x00', ' ').strip('\0')
+        full_name = record['full_name'].replace('\x00', ' ').strip('\0')
+        if initials or full_name: # ensure that at least one field is not empty
+            players[str(i)] = {"initials": initials, "name": full_name}             
+    return json_dumps(players), 200
+
+    
+@add_route("/api/player/update", method="POST", auth=True)
+def app_updatePlayer(request):
+    #TODO if initials and name are empty, we need to delete the scores
```

Comment:
I'm not sure how to delete all of an individual players scores. Currently when a player with a previously used index is created they inherit the deleted players scores

Reply:
ah, I found the snippet I missed, thanks